code/game/machinery/slotmachine.dm

```diff
@@ -118,4 +126,70 @@
 	. = TRUE
 	if(!I.tool_use_check(user, 0))
 		return
+	// This prevents the emagged machine to be moved while someone is playing on it.
+	if(emagged_game_in_progress)
+		return
 	default_unfasten_wrench(user, I)
+
+// Emag behaviour below. When the machine is emagged, it'll stun and anchor its user, spin them, then throw them away.
+// With a chance of "emagged_win_chance", the machine resets its emagged state and throws money at the user.
+/obj/machinery/economy/slot_machine/emag_act(user)
+	if(emagged)
+		to_chat(user, "<span class='notice'>[src] is unresponsive. It is probably already modified.</span>")
+		return
+	playsound(loc, 'sound/effects/sparks4.ogg', 75, 1)
+	emagged = TRUE
+	to_chat(user, "<span class='notice'>You engage the reverse-gripping mechanism on the machine's handle.</span>")
+	log_game("[key_name(user)] emagged [src]")
+
+// The spinning and throwing away is handled here, with a possible call to winning
+/obj/machinery/economy/slot_machine/proc/emagged_spinning(mob/living/user)
+	to_chat(user, "<span class='danger'>As you grip the handle of the machine, it grips back at you, and starts to wildly spin you around!</span>")
+	user.SpinAnimation(speed = 2, loops = 6)
+	emagged_game_in_progress = TRUE
+
+	// Make sure the machine is anchored to avoid people cheesing it
+	if(!anchored)
+		atom_say("Deploying safety bolts.")
+		anchored = TRUE
+
+	// Stun them, there is no escape from this unless you get teleported
+	user.anchored = TRUE
```

Comment:
just use stun here instead of anchoring the user

Reply:
If I stun instead of anchoring, players can drag others away. Do we have any other method of gluing a player to a structure? Because if I just set it to stun, it is not really the machine keeping them hostage, it's just them being unable to act, which is not the same

Reply:
hmm, you'd definitely need to make 100% sure the player gets unanchored after a certain amount of time even if the slot machine is qdel'd

Reply:
The `do_after()` proc takes care of it, its while loop returns false the moment the target stops existing:

https://user-images.githubusercontent.com/33333517/224764661-c126cd39-78ab-4102-995d-5c747ff112fb.mp4